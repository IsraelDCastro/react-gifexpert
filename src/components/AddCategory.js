import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');
  const handleInputChange = (input) => {
    setInputValue(input.target.value)
  }
  const handleSubmit = (form) => {
    form.preventDefault();
    if ( inputValue.trim().length > 2) {
      setCategories( (categories) => [inputValue, ...categories] );
      setInputValue('');
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        value={inputValue}
        onChange={ handleInputChange }
      />
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}


