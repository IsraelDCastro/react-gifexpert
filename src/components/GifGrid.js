import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h4 className="animate__animated animate__fadeIn">{ category }</h4>
      { loading && <p className="animate__animated animate__flash">"Loading..."</p> }
      <div className="card-grid">
        { images.map((image) => (
          <GifGridItem key={image.id} {...image} />
        ))}
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
