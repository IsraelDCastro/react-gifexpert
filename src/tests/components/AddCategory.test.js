import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe("Tests in <AddCategory />", () => {
  const setCategories = jest.fn();
  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("must be rendered correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("must change input text", () => {
    const input = wrapper.find('input');
    const value = "Hello world";
    input.simulate('change', {
      target: {
        value: value
      }
    });
  });

  test("must not post the information with submit", () => {
    const form = wrapper.find('form');
    form.simulate('submit', {
      preventDefault(){}
    })
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("must call call setCategories and clear the input", () => {
    const input = wrapper.find('input');
    const value = "Hello world";
    input.simulate('change', {
      target: {
        value: value
      }
    });

    const form = wrapper.find('form');
    form.simulate('submit', {
      preventDefault(){}
    })

    expect(setCategories).toHaveBeenCalled();
    expect(input.text()).toBe("");
  });
});
