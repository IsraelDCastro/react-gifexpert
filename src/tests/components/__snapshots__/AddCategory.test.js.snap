// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <AddCategory /> must be rendered correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
